### PR TITLE
feat(cli): experimental expo-router flag

### PR DESCRIPTION
**Pull request by frankcalise**
## Please verify the following:

- [ ] `yarn test` **jest** tests pass with new tests, if relevant
- [ ] `yarn lint` **eslint** checks pass with new code, if relevant
- [ ] `yarn format:check` **prettier** checks pass with new code, if relevant
- [ ] `README.md` (or relevant documentation) has been updated with your changes

## Describe your PR
- Adds `expo-router` value to the `--experimental` switch
- Ignites a new app (no demo app available, though) and makes all the updates as per the [Cookbook Recipe](https://ignitecookbook.com/docs/recipes/ExpoRouter/) automatically
- Quick and dirty right now, but I got it out of my brain and it works

## TODO
- [ ] Tests? xd
- [ ] Deploy it to `next` label for in the wild testing
- [ ] Remove the one Reactotron command about resetting nav state
- [ ] Update generators (`model` and `component` as `navigator` and `screen` will vanish) - that being said the cli probably still needs [some more support for sub dir paths on `generate` command](https://github.com/infinitered/ignite/issues/2708)

### CODE REVIEW

File: src/commands/new.ts [hunk 1]
```diff
@@ -768,6 +799,64 @@ module.exports = {
...
       // #endregion
 
+      // #region Expo Router edits
+      /**
+       * instructions mostly adapted from https://ignitecookbook.com/docs/recipes/ExpoRouter
+       * TODO
+       * fix generators
+       * remove the resetNavigation command from reactotronConfig
+       */
+      if (expoRouter) {
```
Comment (jamonholmgren) [inline src/commands/new.ts:809]: I'd like to ask in the CLI walkthrough:

```
Which navigation library?
• React Navigation v7
• Expo Router [experimental]
```

**Review by jamonholmgren — COMMENTED** [review]
Surprisingly simple, but I could see how this would be very useful.

File: src/commands/new.ts (hunk 1 above)
Comment (frankcalise) [reply src/commands/new.ts:809]: Good idea!

File: boilerplate/src/app/index.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+const welcomeLogo = require("../../assets/images/logo.png")
+const welcomeFace = require("../../assets/images/welcome-face.png")
+
+export default observer(function WelcomeScreen() {
+  const $bottomContainerInsets = useSafeAreaInsetsStyle(["bottom"])
+
+  return (
+    <View style={$container}>
+      <View style={$topContainer}>
```
Comment (Jpoliachik) [inline boilerplate/src/app/index.tsx:17]: So I guess we'll need to remember that this screen & _layout are now duplicated, if we ever need to make updates. Not a bad tradeoff, just noteworthy for future maintenance.

Comment (frankcalise) [reply boilerplate/src/app/index.tsx:17]: The `_layout.tsx` isn't but yeah `WelcomeScreen.tsx` -> `index.tsx` is, I don't love it. But it wasn't as simple as a rename. 

I thought coding more regex replace stuff was uglier to maintain over just moving it. We don't change the welcome screen that often but yeah we could cook up something else

**Comment by frankcalise** [comment]
Closing in favor of #2727, feel free to continue discussion there. This will be part of the v10 release

**Pull request closed without merging**